### PR TITLE
[pt2] add meta for `ormqr`

### PR DESCRIPTION
**Pull request by nkaretnikov**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #106278

### CODE REVIEW

File: torch/_meta_registrations.py [hunk 1]
```diff
@@ -1150,6 +1150,98 @@ def _linalg_det_meta(A):
...
+    torch._check(
+        tau.shape[-1] <= input.shape[-1],
+        lambda: "torch.ormqr: tau.shape[-1] must be less than or equal to input.shape[-1]",
+    )
+
+    torch._check(
+        input.ndim - tau.ndim == 1,
+        lambda: (
+            f"torch.ormqr: Expected tau to have one dimension less than input, "
+            f"but got tau.ndim equal to {tau.ndim} and {input.ndim} is equal to {input.ndim}"
```
Comment (nkaretnikov) [inline torch/_meta_registrations.py:1188]: Will fix: first input.ndim shouldn't be in curly brackets.

**Comment by nkaretnikov** [comment]
From the list in https://github.com/pytorch/pytorch/issues/105105

File: torch/_meta_registrations.py [hunk 2]
```diff
@@ -1150,6 +1150,98 @@ def _linalg_det_meta(A):
...
+
+    checkSameDevice("torch.ormqr", tau, input, "tau")
+    checkSameDevice("torch.ormqr", other, input, "other")
+
+    return torch.empty_strided(
+        size=other.shape,
+        stride=make_contiguous_strides_for(other.shape, row_major=False),
+        dtype=other.dtype,
+        device=other.device,
+    )
```
Comment (nkaretnikov) [inline torch/_meta_registrations.py:1242]: This is very similar to linalg_householder_product meta.

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**